edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +224,30 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let lock = if let ConcurrencyThrottling::Limited(limit) = concurrency {
+            Some((Arc::new(AtomicU32::new(limit)), limit))
```

Comment:
Interesting way to keep track of the limit... is this a pattern?

Reply:
I improvise it :). This was the code on the first PR version:
![image](https://user-images.githubusercontent.com/33438817/124246958-44f9a500-dad6-11eb-8f21-abfbbe015302.png)
I changed it to decrement instead of increment. I also added a tuple, to keep the original limit in memory. See previous comments:
![image](https://user-images.githubusercontent.com/33438817/124247327-a3bf1e80-dad6-11eb-975e-449396c98787.png)
and
![image](https://user-images.githubusercontent.com/33438817/124247662-ef71c800-dad6-11eb-9d54-c4a50886d9b9.png)